r2/r2/models/subreddit.py

```diff
@@ -540,6 +540,33 @@ def remove_moderator(self, user, **kwargs):
         hook = hooks.get_hook("subreddit.remove_moderator")
         hook.call(subreddit=self, user=user)
 
+        if user == self.top_moderator_account and len(self.moderators) > 1:
+            from r2.models import ModAction
+            fullperms = self.moderator_permission_class.loads('+all')
+            # cycle through each available moderator
+            for user_id in self.moderators[1:]:
+                successor = Account._byID(user_id)
+                # skip deleted / shadowbanned users
+                if successor._deleted or successor._spam:
```

Comment:
I think we had probably better not skip `._spam` users for now. We still more or less consider them "valid mods" for most things (they're still in the mod list and such), so if we skip over them there's a risk of a weird case where a temporarily-shadowbanned user gets skipped as a successor, and full perms actually pass to the third (or lower) mod instead of the second. I think just skipping deleted ones is best to avoid that case for now.